File: client/custom_open/src/feedback/sql_queries.js
import sqlite3 from 'sqlite3';
import { Paths } from './config.js';

function DBConnect(query, userId = null) {
    return new Promise((resolve, reject) => {
        const db = new sqlite3.Database(Paths.DB_FILE, (err) => {
            if (err) {
                reject(err);
            }
        });

        db.all(query, userId ? userId : [], (err, rows) => {
            if (err) {
                reject(err);
            } else {
                resolve(rows);
            }
            db.close();
        });
    });
}

export { DBConnect };

File: services/document-manager/src/services/feedback/sql_queries.py
import sqlite3
from utils.config import Paths


def DBConnect(query, user_id=""):
    connection = sqlite3.connect(Paths.DB_FILE)
    cursor = connection.cursor()
    cursor.execute(query, user_id)
    result = cursor.fetchall()
    connection.close()
    return result
